src/java/org/apache/cassandra/db/tries/MergeTrie.java

```diff
@@ -149,6 +151,13 @@ public Direction direction()
             return direction;
         }
 
+        @Override
+        public ByteComparable.Version byteComparableVersion()
+        {
+            assert c1.byteComparableVersion() == c2.byteComparableVersion();
```

Comment:
Nit: I'd add a message for the assert that logs the 2 versions (I've been burned a bunch of time with assertions failing in prod with no clue why and being frustrated as to how the assertion failed exactly).